lib/src/pool/pool_impl.dart

```diff
@@ -177,7 +177,7 @@ class PoolImplementation<L> implements Pool<L> {
   Future<_PoolConnection> _selectOrCreate(
       Endpoint endpoint, ResolvedConnectionSettings settings) async {
     final oldc =
-        _connections.firstWhereOrNull((c) => c._mayReuse(endpoint, settings));
+        _connections.firstWhereOrNull((c) => c._mayReuse(endpoint, settings) && c.isOpen);
```

Comment:
How about putting that condition in the `_mayReuse` method? Also, if it is not a hassle, could you please add a note in the changelog?